### PR TITLE
feat: bring your own ECS + LB

### PR DESCRIPTION
**Pull request by endrec**
## Description
A set of opinionated changes to allow one to use their existing ECS and LB.

## Motivation and Context
We wanted to deploy multiple atlantis servers (one per environment), but use the same ECS cluster and load balancer.
This change depends on https://github.com/terraform-aws-modules/terraform-aws-alb/pull/227.

## Breaking Changes
Removed input_allow_repo_config variable, as the setting is deprecated in atlantis.

## How Has This Been Tested?
We run these changes ourselves.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -192,13 +196,26 @@ module "vpc" {
...
+data "aws_lb_listener" "https_listener" {
+  load_balancer_arn = local.alb_arn
+  port              = 443
+}
+
 module "alb" {
-  source  = "terraform-aws-modules/alb/aws"
-  version = "v6.5.0"
+  source = "git@github.com:endrec/terraform-aws-alb.git?ref=byo"
+  #  source  = "terraform-aws-modules/alb/aws"
```
Comment (bryantbiggs) [inline main.tf:211]: we should leave this as the ALB module managed under this organization

**Review by bryantbiggs — CHANGES_REQUESTED** [review]
we will need to stick with the ALB module provided under the `terraform-aws-modules` organization

File: main.tf (hunk 1 above)
Comment (endrec) [reply main.tf:211]: Sure we should, but I don't know the version number till terraform-aws-modules/terraform-aws-alb#227 is released.

**Comment by endrec** [comment]
This PR depends on a change which won't get merge: 
https://github.com/terraform-aws-modules/terraform-aws-alb/pull/227#issuecomment-1009891950

**Pull request closed without merging**